enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/EnterpriseRuntimeBuilder.scala

```diff
@@ -29,15 +29,25 @@ import org.neo4j.cypher.internal.frontend.v3_3.phases.{Do, If, Transformer}
 
 object EnterpriseRuntimeBuilder extends RuntimeBuilder[Transformer[EnterpriseRuntimeContext, LogicalPlanState, CompilationState]] {
   def create(runtimeName: Option[RuntimeName], useErrorsOverWarnings: Boolean): Transformer[EnterpriseRuntimeContext, LogicalPlanState, CompilationState] = {
+
+    def pickInterpretedExecutionPlan() =
+      BuildEnterpriseInterpretedExecutionPlan andThen
+        If[EnterpriseRuntimeContext, LogicalPlanState, CompilationState](_.maybeExecutionPlan.isEmpty) {
+          BuildInterpretedExecutionPlan
+        }
+
     runtimeName match {
       case None =>
         BuildCompiledExecutionPlan andThen
           If[EnterpriseRuntimeContext, LogicalPlanState, CompilationState](_.maybeExecutionPlan.isEmpty) {
-            BuildEnterpriseInterpretedExecutionPlan
+            pickInterpretedExecutionPlan()
           }
 
       case Some(InterpretedRuntimeName) =>
-        BuildEnterpriseInterpretedExecutionPlan
+        BuildInterpretedExecutionPlan
+
+      case Some(EnterpriseInterpretedRuntimeName) =>
```

Comment:
When the user has specified enterprise runtime, we should either warn or fail if that is not possible to achieve